urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/publishers/FillMessage.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * FillMessage.java
+ */
+package publishers;
+
+import constants.BookSide;
+import price.Price;
+
+/**
+ * This class encapsulates data related to the fill (trade) of an order or
+ * quote-side.
+ *
+ * @author Kevin Patrick Westropp
+ * @version StockExVirtuoso.Version.1.0
+ */
+public class FillMessage extends Messenger implements Comparable<FillMessage>, Message {
+
+    /**
+     * Public Construction method to create a new Fill Message.
+     *
+     * @param userName of user
+     * @param productSymbol of stock
+     * @param p price object
+     * @param newVolume
+     * @param details fill details
+     * @param s side of book
+     * @param newId of order/quote
+     * @throws Exception
+     */
+    public FillMessage(String userName, String productSymbol, Price p, int newVolume, String details, BookSide s, String newId) throws Exception {
+        super(userName, productSymbol, p, newVolume, details, s, newId);
+    }
+
+    /**
+     * Public compare To method which compares the prices of two FillMessages.
+     *
+     * @param fm FillMessage passed in
+     * @return integer from call to compareTo on price, 0 if equal, -1 if
+     * exception or not equal
+     */
+    @Override
+    public int compareTo(FillMessage fm) {
+        try {
+            Price currentPrice = getPrice();
+            Price comparePrice = fm.getPrice();
+            return currentPrice.compareTo(comparePrice);
+        } catch (Exception ex) {
+            System.out.print(ex.getMessage());
+            return -1;
+        }
+    }
+
+    /**
+     * Public To string method which returns a string of the values of this
+     * message.
+     *
+     * @return String of values of the FillMessage
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("User: %s, Product: %s, Fill Price: %s,", getUser(), getProduct(), getPriceString()));
+        sb.append(String.format(" Fill Volume: %s, Details: %s, Side: %s", getVolume(), getDetails(), getSide()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getVolume&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 63. Consider formatting this data with &#39;%d&#39; instead.